### ISSUE TITLE
Constants.ts may be designed incompatible with mxConstants when migrating from mxGraph

### ISSUE THREAD

**Issue by rivenww** [maxGraph/maxgraph#192]
Hello guys, here are some suggestions about _**Constants.ts**_. I have used mxgraph for a long time and I receive a mail from my my former colleagues.
When we used mxgraph previously, we know that the mxConstants is modifiable, be like a kind of global varaibles collection.
So, we can simply modify one of its value and change the global apperance.
For example, when I want the stroke width of the edges to be 3px in my graph, I should change the value in mxConstants.

However, in the new version, we define all the values separately and they are all defined with const keyword in typescript. This isn't the original intention of mxConstants.
It should be a const object, with values inside the object. So that the users won't change the reference of the constant object but users can change the value in it.
The Constants.ts make maxGraph is very difficult for developers to customize.
Thanks, and if anyone agree with my opinions, please let me know and maybe I can make a pull request for this repo.

**Comment by eRedekopp** [comment]
I agree with this. For example, I want to make arrow handles slightly larger by modifying the `HANDLE_SIZE` constant, but instead I had to make a whole new EdgeHandler subclass and override the `createHandleShape` method which seems unnecessarily difficult.

**Comment by tbouffard** [comment]
Hi all,

This topic has been discussed extensively, as seen in issues like #83 and #111, and I completely agree - it’s a pain point.

Historically, `mxConstants` in mxGraph stored various types of data:

1. **True constants**: These were invariants not meant to be updated, like `NS_XLINK`, `NS_SVG`, `PAGE_FORMAT_A4_PORTRAIT`, and `PAGE_FORMAT_LETTER_LANDSCAPE`.
2. **Style reference keys**: Strings starting with `STYLE_`, which could technically be updated but came with potential side effects (e.g., impacting the encode/decode functionality).
3. **Global configuration values**: These were modifiable and allowed users to tweak mxGraph’s default behavior.

Given this mix of responsibilities, the term "constants" wasn't entirely accurate for what `mxConstants` was actually storing. 

During the migration to maxGraph, contributors likely interpreted the "constants" name literally and decided to make all values immutable in the `constants` namespace. This aligned with modern coding practices but reduced flexibility.

### Current and Future Updates

To address this, I’ve recently begun making some keys configurable again. While the `constants` namespace remains, new global configuration objects have been introduced to separate truly immutable values from those meant for customization. This will be extended to all configurable values in future updates.

The updated documentation explains the new configuration objects:

- [Global Configuration Documentation](https://maxgraph.github.io/maxGraph/docs/usage/global-configuration/)
- [Migration Guide for `mxConstants`](https://maxgraph.github.io/maxGraph/docs/usage/migrate-from-mxgraph#mxconstants)

### Examples of New Configurable Objects

- **Handler Configurations**: `VertexHandlerConfig`, `EdgeHandlerConfig`, and `HandleConfig` now allow customization of settings previously modified via the prototype in mxGraph. Initial features were introduced in https://github.com/maxGraph/maxGraph/releases/tag/v0.14.0
- **Style Defaults**: Starting with version 0.14.0, the `StyleDefaultsConfig` object lets you adjust default style values, such as shadow properties.

### Work in Progress

There’s still more to do. For example, some default style configurations like `RECTANGLE_ROUNDING_FACTOR` remain immutable, and no straightforward way is provided to update them via a style property. This is on my radar for future improvements.
